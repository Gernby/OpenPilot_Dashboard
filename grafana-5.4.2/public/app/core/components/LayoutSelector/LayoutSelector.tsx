import React, { SFC } from 'react';

export type LayoutMode = LayoutModes.Grid | LayoutModes.List;

export enum LayoutModes {
  Grid = 'grid',
  List = 'list',
}

interface Props {
  mode: LayoutMode;
  onLayoutModeChanged: (mode: LayoutMode) => {};
}

const LayoutSelector: SFC<Props> = props => {
  const { mode, onLayoutModeChanged } = props;
  return (
    <div className="layout-selector">
      <button
        onClick={() => {
          onLayoutModeChanged(LayoutModes.List);
        }}
        className={mode === LayoutModes.List ? 'active' : ''}
      >
        <i className="fa fa-list" />
      </button>
      <button
        onClick={() => {
          onLayoutModeChanged(LayoutModes.Grid);
        }}
        className={mode === LayoutModes.Grid ? 'active' : ''}
      >
        <i className="fa fa-th" />
      </button>
    </div>
  );
};

export default LayoutSelector;
