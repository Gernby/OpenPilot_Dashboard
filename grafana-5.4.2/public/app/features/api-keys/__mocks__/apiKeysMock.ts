import { ApiKey, OrgRole } from 'app/types';

export const getMultipleMockKeys = (numberOfKeys: number): ApiKey[] => {
  const keys: ApiKey[] = [];
  for (let i = 1; i <= numberOfKeys; i++) {
    keys.push({
      id: i,
      name: `test-${i}`,
      role: OrgRole.Viewer,
    });
  }

  return keys;
};

export const getMockKey = (): ApiKey => {
  return {
    id: 1,
    name: 'test',
    role: OrgRole.Admin,
  };
};
