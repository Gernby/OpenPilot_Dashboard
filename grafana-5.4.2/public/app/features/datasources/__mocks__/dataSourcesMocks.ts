import { DataSource } from 'app/types';

export const getMockDataSources = (amount: number): DataSource[] => {
  const dataSources = [];

  for (let i = 0; i <= amount; i++) {
    dataSources.push({
      access: '',
      basicAuth: false,
      database: `database-${i}`,
      id: i,
      isDefault: false,
      jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
      name: `dataSource-${i}`,
      orgId: 1,
      password: '',
      readOnly: false,
      type: 'cloudwatch',
      typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
      url: '',
      user: '',
    });
  }

  return dataSources;
};

export const getMockDataSource = (): DataSource => {
  return {
    access: '',
    basicAuth: false,
    database: '',
    id: 13,
    isDefault: false,
    jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
    name: 'gdev-cloudwatch',
    orgId: 1,
    password: '',
    readOnly: false,
    type: 'cloudwatch',
    typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
    url: '',
    user: '',
  };
};
