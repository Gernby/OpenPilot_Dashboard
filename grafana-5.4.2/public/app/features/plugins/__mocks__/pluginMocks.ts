import { Plugin } from 'app/types';

export const getMockPlugins = (amount: number): Plugin[] => {
  const plugins = [];

  for (let i = 0; i <= amount; i++) {
    plugins.push({
      defaultNavUrl: 'some/url',
      enabled: false,
      hasUpdate: false,
      id: `${i}`,
      info: {
        author: {
          name: 'Grafana Labs',
          url: 'url/to/GrafanaLabs',
        },
        description: 'pretty decent plugin',
        links: ['one link'],
        logos: { small: 'small/logo', large: 'large/logo' },
        screenshots: `screenshot/${i}`,
        updated: '2018-09-26',
        version: '1',
      },
      latestVersion: `1.${i}`,
      name: `pretty cool plugin-${i}`,
      pinned: false,
      state: '',
      type: '',
    });
  }

  return plugins;
};

export const getMockPlugin = () => {
  return {
    defaultNavUrl: 'some/url',
    enabled: false,
    hasUpdate: false,
    id: '1',
    info: {
      author: {
        name: 'Grafana Labs',
        url: 'url/to/GrafanaLabs',
      },
      description: 'pretty decent plugin',
      links: ['one link'],
      logos: { small: 'small/logo', large: 'large/logo' },
      screenshots: 'screenshot/1',
      updated: '2018-09-26',
      version: '1',
    },
    latestVersion: '1',
    name: 'pretty cool plugin 1',
    pinned: false,
    state: '',
    type: '',
  };
};
