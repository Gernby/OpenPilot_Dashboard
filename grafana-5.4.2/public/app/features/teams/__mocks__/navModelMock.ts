export const getMockNavModel = (pageName: string) => {
  return {
    node: {
      active: false,
      icon: 'gicon gicon-team',
      id: `team-${pageName}-2`,
      text: `${pageName}`,
      url: 'org/teams/edit/2/members',
      parentItem: {
        img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
        id: 'team-2',
        subTitle: 'Manage members & settings',
        url: '',
        text: 'test1',
        breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
        children: [
          {
            active: false,
            icon: 'gicon gicon-team',
            id: 'team-members-2',
            text: 'Members',
            url: 'org/teams/edit/2/members',
          },
          {
            active: false,
            icon: 'fa fa-fw fa-sliders',
            id: 'team-settings-2',
            text: 'Settings',
            url: 'org/teams/edit/2/settings',
          },
        ],
      },
    },
    main: {
      img: '/avatar/b5695b61c91d13e7fa2fe71cfb95de9b',
      id: 'team-2',
      subTitle: 'Manage members & settings',
      url: '',
      text: 'test1',
      breadcrumbs: [{ title: 'Teams', url: 'org/teams' }],
      children: [
        {
          active: true,
          icon: 'gicon gicon-team',
          id: 'team-members-2',
          text: 'Members',
          url: 'org/teams/edit/2/members',
        },
        {
          active: false,
          icon: 'fa fa-fw fa-sliders',
          id: 'team-settings-2',
          text: 'Settings',
          url: 'org/teams/edit/2/settings',
        },
      ],
    },
  };
};
