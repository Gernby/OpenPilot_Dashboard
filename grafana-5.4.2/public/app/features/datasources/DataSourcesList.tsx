import React, { PureComponent } from 'react';
import classNames from 'classnames/bind';
import DataSourcesListItem from './DataSourcesListItem';
import { DataSource } from 'app/types';
import { LayoutMode, LayoutModes } from '../../core/components/LayoutSelector/LayoutSelector';

export interface Props {
  dataSources: DataSource[];
  layoutMode: LayoutMode;
}

export class DataSourcesList extends PureComponent<Props> {
  render() {
    const { dataSources, layoutMode } = this.props;

    const listStyle = classNames({
      'card-section': true,
      'card-list-layout-grid': layoutMode === LayoutModes.Grid,
      'card-list-layout-list': layoutMode === LayoutModes.List,
    });

    return (
      <section className={listStyle}>
        <ol className="card-list">
          {dataSources.map((dataSource, index) => {
            return <DataSourcesListItem dataSource={dataSource} key={`${dataSource.id}-${index}`} />;
          })}
        </ol>
      </section>
    );
  }
}

export default DataSourcesList;
