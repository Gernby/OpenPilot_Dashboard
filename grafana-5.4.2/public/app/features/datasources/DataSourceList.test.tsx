import React from 'react';
import { shallow } from 'enzyme';
import DataSourcesList from './DataSourcesList';
import { getMockDataSources } from './__mocks__/dataSourcesMocks';
import { LayoutModes } from '../../core/components/LayoutSelector/LayoutSelector';

const setup = () => {
  const props = {
    dataSources: getMockDataSources(3),
    layoutMode: LayoutModes.Grid,
  };

  return shallow(<DataSourcesList {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
