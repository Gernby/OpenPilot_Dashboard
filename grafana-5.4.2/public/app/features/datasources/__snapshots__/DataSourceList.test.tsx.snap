// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<section
  className="card-section card-list-layout-grid"
>
  <ol
    className="card-list"
  >
    <DataSourcesListItem
      dataSource={
        Object {
          "access": "",
          "basicAuth": false,
          "database": "database-0",
          "id": 0,
          "isDefault": false,
          "jsonData": Object {
            "authType": "credentials",
            "defaultRegion": "eu-west-2",
          },
          "name": "dataSource-0",
          "orgId": 1,
          "password": "",
          "readOnly": false,
          "type": "cloudwatch",
          "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
          "url": "",
          "user": "",
        }
      }
      key="0-0"
    />
    <DataSourcesListItem
      dataSource={
        Object {
          "access": "",
          "basicAuth": false,
          "database": "database-1",
          "id": 1,
          "isDefault": false,
          "jsonData": Object {
            "authType": "credentials",
            "defaultRegion": "eu-west-2",
          },
          "name": "dataSource-1",
          "orgId": 1,
          "password": "",
          "readOnly": false,
          "type": "cloudwatch",
          "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
          "url": "",
          "user": "",
        }
      }
      key="1-1"
    />
    <DataSourcesListItem
      dataSource={
        Object {
          "access": "",
          "basicAuth": false,
          "database": "database-2",
          "id": 2,
          "isDefault": false,
          "jsonData": Object {
            "authType": "credentials",
            "defaultRegion": "eu-west-2",
          },
          "name": "dataSource-2",
          "orgId": 1,
          "password": "",
          "readOnly": false,
          "type": "cloudwatch",
          "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
          "url": "",
          "user": "",
        }
      }
      key="2-2"
    />
    <DataSourcesListItem
      dataSource={
        Object {
          "access": "",
          "basicAuth": false,
          "database": "database-3",
          "id": 3,
          "isDefault": false,
          "jsonData": Object {
            "authType": "credentials",
            "defaultRegion": "eu-west-2",
          },
          "name": "dataSource-3",
          "orgId": 1,
          "password": "",
          "readOnly": false,
          "type": "cloudwatch",
          "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
          "url": "",
          "user": "",
        }
      }
      key="3-3"
    />
  </ol>
</section>
`;
